### ISSUE TITLE
[2022/03/25 11:15:53] [engine] caught signal (SIGSEGV)

### ISSUE THREAD

**Issue by nickytd** [fluent/fluent-bit#5185]
## Bug Report

**Describe the bug**
An error occur when switching from es output plugin to opensearch 
The plugin configurations uses TLS and the error stack is as follows
```
[2022/03/25 11:15:53] [engine] caught signal (SIGSEGV)
#0  0x7fad57bb82b9      in  ???() at ???:0
#1  0x7fad57b21039      in  ???() at ???:0
#2  0x7fad57cac819      in  ???() at ???:0
#3  0x7fad57cbc4ad      in  ???() at ???:0
#4  0x56266835fdb2      in  tls_session_destroy() at src/tls/openssl.c:385
#5  0x562668360b89      in  flb_tls_session_destroy() at src/tls/flb_tls.c:439
#6  0x562668360b43      in  flb_tls_session_create() at src/tls/flb_tls.c:429
#7  0x56266836b6d1      in  flb_io_net_connect() at src/flb_io.c:109
#8  0x56266834ab10      in  create_conn() at src/flb_upstream.c:560
#9  0x56266834b00c      in  flb_upstream_conn_get() at src/flb_upstream.c:705
#10 0x56266843ecc3      in  get_meta_info_from_request() at plugins/filter_kubernetes/kube_meta.c:370
#11 0x56266843f3a3      in  get_api_server_info() at plugins/filter_kubernetes/kube_meta.c:492
#12 0x562668441a30      in  get_and_merge_meta() at plugins/filter_kubernetes/kube_meta.c:1352
#13 0x562668442537      in  flb_kube_meta_get() at plugins/filter_kubernetes/kube_meta.c:1560
#14 0x56266843cc37      in  cb_kube_filter() at plugins/filter_kubernetes/kubernetes.c:469
#15 0x56266833063e      in  flb_filter_do() at src/flb_filter.c:123
#16 0x56266836ac28      in  input_chunk_append_raw() at src/flb_input_chunk.c:1478
#17 0x56266836af0b      in  flb_input_chunk_append_raw() at src/flb_input_chunk.c:1592
#18 0x562668395f37      in  ml_flush_callback() at plugins/in_tail/tail_file.c:838
#19 0x562668356da9      in  flb_ml_flush_stream_group() at src/multiline/flb_ml.c:1110
#20 0x562668355383      in  package_content() at src/multiline/flb_ml.c:333
#21 0x562668355808      in  process_append() at src/multiline/flb_ml.c:499
#22 0x562668355d4c      in  ml_append_try_parser() at src/multiline/flb_ml.c:659
#23 0x562668355f58      in  flb_ml_append() at src/multiline/flb_ml.c:725
#24 0x5626683950f0      in  process_content() at plugins/in_tail/tail_file.c:434
#25 0x56266839782b      in  flb_tail_file_chunk() at plugins/in_tail/tail_file.c:1315
#26 0x56266838f91b      in  in_tail_collect_event() at plugins/in_tail/tail.c:310
#27 0x562668390b2a      in  tail_fs_event() at plugins/in_tail/tail_fs_inotify.c:267
#28 0x56266833013b      in  flb_input_collector_fd() at src/flb_input.c:1203
#29 0x56266834409a      in  flb_engine_handle_event() at src/flb_engine.c:439
#30 0x56266834409a      in  flb_engine_start() at src/flb_engine.c:761
#31 0x562668323a14      in  flb_lib_worker() at src/flb_lib.c:626
#32 0x7fad57df2ea6      in  ???() at ???:0
#33 0x7fad576d2dee      in  ???() at ???:0
#34 0xffffffffffffffff  in  ???() at ???:0
```
On opensearch node side there is corresponding java exception
```
"Caused by: javax.net.ssl.SSLHandshakeException: Insufficient buffer remaining for AEAD cipher fragment (2). Needs to be more than tag size (16)",
"at sun.security.ssl.Alert.createSSLException(Alert.java:131) ~[?:?]",
"at sun.security.ssl.TransportContext.fatal(TransportContext.java:352) ~[?:?]",
"at sun.security.ssl.TransportContext.fatal(TransportContext.java:295) ~[?:?]",
"at sun.security.ssl.TransportContext.fatal(TransportContext.java:290) ~[?:?]",
"at sun.security.ssl.SSLTransport.decode(SSLTransport.java:134) ~[?:?]",
"at sun.security.ssl.SSLEngineImpl.decode(SSLEngineImpl.java:681) ~[?:?]",
"at sun.security.ssl.SSLEngineImpl.readRecord(SSLEngineImpl.java:636) ~[?:?]",
"at sun.security.ssl.SSLEngineImpl.unwrap(SSLEngineImpl.java:454) ~[?:?]",
"at sun.security.ssl.SSLEngineImpl.unwrap(SSLEngineImpl.java:433) ~[?:?]",
"at javax.net.ssl.SSLEngine.unwrap(SSLEngine.java:637) ~[?:?]",
"at io.netty.handler.ssl.SslHandler$SslEngineType$3.unwrap(SslHandler.java:295) ~[netty-handler-4.1.73.Final.jar:4.1.73.Final]",
```
**To Reproduce**
Setup opensearch with https endpoint and self-signed CA
**Your Environment**
opensearch 1.3.0
fluentbit 1.9.0

The very same fluentbit configuration setup but with "es" (instead of "opensearch) output plugin works as expected

**Comment by nickytd** [comment]
issue is solved with opensearch 1.3.1 and fluent-bit 1.9.2

**Comment by tooptoop4** [comment]
i got this in 2.2.0:

`[2025/01/redact] [engine] caught signal (SIGSEGV)
#0  0x557d7ff41047      in  search_metadata_in_items() at plugins/filter_kubernetes/kube_meta.c:776
#1  0x557d7ff4142b      in  search_item_in_items() at plugins/filter_kubernetes/kube_meta.c:845
#2  0x557d7ff417f4      in  merge_meta() at plugins/filter_kubernetes/kube_meta.c:962
#3  0x557d7ff42d30      in  get_and_merge_meta() at plugins/filter_kubernetes/kube_meta.c:1356
#4  0x557d7ff43a12      in  flb_kube_meta_get() at plugins/filter_kubernetes/kube_meta.c:1558
#5  0x557d7ff36c1e      in  cb_kube_filter() at plugins/filter_kubernetes/kubernetes.c:552
#6  0x557d7f9e2581      in  flb_filter_do() at src/flb_filter.c:158
#7  0x557d7f9d81c6      in  input_chunk_append_raw() at src/flb_input_chunk.c:1533
#8  0x557d7f9d8ada      in  flb_input_chunk_append_raw() at src/flb_input_chunk.c:1845
#9  0x557d7fac18db      in  input_log_append() at src/flb_input_log.c:71
#10 0x557d7fac1966      in  flb_input_log_append() at src/flb_input_log.c:90
#11 0x557d7fbf6c65      in  ml_stream_buffer_flush() at plugins/in_tail/tail_file.c:410
#12 0x557d7fbf7267      in  process_content() at plugins/in_tail/tail_file.c:622
#13 0x557d7fbf9a29      in  flb_tail_file_chunk() at plugins/in_tail/tail_file.c:1415
#14 0x557d7fbe398f      in  in_tail_collect_event() at plugins/in_tail/tail.c:337
#15 0x557d7fbe84b8      in  tail_fs_event() at plugins/in_tail/tail_fs_inotify.c:276
#16 0x557d7f9d1a57      in  flb_input_collector_fd() at src/flb_input.c:1948
#17 0x557d7fa0fba6      in  flb_engine_handle_event() at src/flb_engine.c:543
#18 0x557d7fa0fba6      in  flb_engine_start() at src/flb_engine.c:909
#19 0x557d7f9b330a      in  flb_lib_worker() at src/flb_lib.c:638
#20 0x7fd7b5de1ea6      in  ???() at ???:0
#21 0x7fd7b5695a2e      in  ???() at ???:0
#22 0xffffffffffffffff  in  ???() at ???:0`
